src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java

```diff
@@ -79,6 +81,28 @@ public ZipFileStatus getStatusFor(String zipFileName) {
         return zipFileStatusList;
     }
 
+    public CcdIdStatus getStatusByCcdId(String ccdId) {
+
+        List<Envelope> envelopes = envelopeRepo.findByCcdId(ccdId);
+        if (envelopes.size() > 0) {
+            String zipFileName = envelopes.get(0).getZipFileName();
+            List<ProcessEvent> events = eventRepo.findByZipFileName(zipFileName);
+            ZipFileStatus zipFileStatus = new ZipFileStatus(zipFileName,
+                                          envelopes.stream().map(this::mapEnvelope).collect(toList()),
+                                          events.stream().map(this::mapEvent).collect(toList()));
+            return new CcdIdStatus(
+                ccdId,
+                zipFileStatus
+            );
+        }
+        return new CcdIdStatus(
+        "",
```

Comment:
Should be `ccdId`

Reply:
Yes, makes sense, it will return ccdid with empty zipfilestatus now.